subprojects/dependency-management/src/testFixtures/groovy/org/gradle/integtests/fixtures/publish/ModuleVersionSpec.groovy

```diff
@@ -129,11 +129,20 @@ class ModuleVersionSpec {
     }
 
     void variant(String name, @DelegatesTo(value = VariantMetadataSpec, strategy = Closure.DELEGATE_FIRST) Closure<?> spec) {
-        def variant = new VariantMetadataSpec(name)
+        def variant = variants.find { it.name == name }
```

Comment:
This changes are not exercised by the PR. If we were to put it in 6.6.1, I would drop them from the PR.

Reply:
Indeed I wrote that as part of trying to write a test case to exercise this problem.